test/Test/MockApi.hs

```diff
@@ -75,27 +79,43 @@ data ExpectationRepetition
   | Always
   deriving (Eq, Ord, Show)
 
+data ExpectationRequestType
+  = ExpectingExactRequest
+  | ExpectingAnyRequest
+  deriving (Eq, Ord, Show)
+
 -- | The result of a call can be either a value or a diagnostic failure
 newtype ApiResult a = ApiResult (Either ApiFail a)
 
 newtype ApiFail = ApiFail {unApiFail :: Text}
 
 -- | An expectation of an API call made up of the request and response.
 data ApiExpectation where
-  ApiExpectation :: ExpectationRepetition -> FossaApiClientF a -> ApiResult a -> ApiExpectation
+  ApiExpectation :: ExpectationRepetition -> ExpectationRequestType -> FossaApiClientF a -> ApiResult a -> ApiExpectation
 
 -- | Create an expectation that will only be satisfied once.
 returnsOnce :: Has MockApi sig m => FossaApiClientF a -> a -> m ()
 returnsOnce req resp =
   -- ApiExpectation Once req (Return resp)
   send $ MockApiOnce req resp
 
+-- | Create an expectation that will only be satisfied once where we do not check the value of the req
+returnsOnceForAnyRequest :: Has MockApi sig m => FossaApiClientF a -> a -> m ()
```

Comment:
Would `returnsOnceForAnyRequestBody` and `alwaysReturnsForAnyRequestBody` be better names, or is that just too wordy?

Reply:
I think it might be getting too wordy.  Maybe we can think back to what we would want to read and something will jump out at us.  "request returnsOnceForSimilarRequests response"? "request returnsOnceIgnoringContents responce"?

I'll leave it up to you.  I think the ergonomics are minor and very easy to change.

Reply:
I'm going to leave it as is for now, since it will be an easy search-and-replace if we come up with a better name in the future